src/compiler/crystal/tools/unreachable.cr

```diff
@@ -199,11 +208,10 @@ module Crystal
 
       check_def(previous) if previous && !a_def.calls_previous_def?
 
-      return if @used_def_locations.includes?(a_def.location)
-
-      check_def(previous) if previous && a_def.calls_previous_def?
+      tally = @used_def_locations[a_def.location]
+      @tallies[a_def] = tally
 
-      @defs << a_def
+      check_def(previous) if previous && a_def.calls_previous_def? if tally == 0
```

Comment:
Are these the same?

```suggestion
      check_def(previous) if previous && a_def.calls_previous_def? && tally == 0
```

Reply:
Yeah, should be 😅

Reply:
I mean the order of execution changes, but there are no side effects on these.